cpp/src/gandiva/function_registry_datetime.cc

```diff
@@ -26,7 +26,7 @@ namespace gandiva {
       DATE_TYPES(INNER, name##Decade, {}), DATE_TYPES(INNER, name##Year, {"year"}),    \
       DATE_TYPES(INNER, name##Quarter, {}), DATE_TYPES(INNER, name##Month, {"month"}), \
       DATE_TYPES(INNER, name##Week, ({"weekofyear", "yearweek"})),                     \
-      DATE_TYPES(INNER, name##Day, ({"day", "dayofmonth"})),                           \
+      DATE_TYPES(INNER, name##Day, ({"day", "dayofmonth", "DAYOFMONTH"})),             \
```

Comment:
The Gandiva functions are case-insensitive, so you do not need to add DAYOFMONTH.